tools/tpm2_ptool.py

```diff
@@ -1030,27 +1074,211 @@ def __call__(self, args):
+                {  CKA_CLASS           : CKO_PUBLIC_KEY  },
+                {  CKA_ID              : id              },
+                {  CKA_MODULUS         : y['rsa']        },
+                {  CKA_PUBLIC_EXPONENT : 65537           },
+            ]
+
+            mech = [
+                { CKM_RSA_PKCS_OAEP : "" },
+            ]
+
+        elif alg.startswith('aes'):
+            attrs = [
+                { CKA_CLASS    : CKO_SECRET_KEY },
+                { CKA_KEY_TYPE : CKK_AES        }
+            ]
+
+            mech = [
+                { CKM_AES_CBC : "" },
+            ]
+
+        # Add keylabel for ALL objects if set
+        if keylabel is not None:
+            attrs.append({CKA_LABEL : keylabel})
+
+        # Now get the secondary object from db
+        sobj = db.getsecondary(token['id'])
+
+        # Store to database
+        id = db.addtertiary(sobj['id'], tertiarypriv, tertiarypub, encobjauth, attrs, mech)
+
+        # if the keylabel is not set, use the tertiary object id as the keylabel
+        # Normally we would use a transaction to make this atomic, but Pythons
+        # sqlite3 transaction handling is quite odd. So when the keylabel is None, just insert
+        # into the db without that attribute, retrieve the primary key, and then issue an
+        # update. A possible race exists if someone is looking for the key by label between
+        # these operations.
+        # See:
+        #   - https://stackoverflow.com/questions/107005/predict-next-auto-inserted-row-id-sqlite
+        if keylabel is None:
+            keylabel = str(id)
+            attrs.append({CKA_LABEL : keylabel})
+            db.updatetertiaryattrs(id, attrs)
+
+        db.commit()
+
+
+    def __call__(self, args):
+        path = args['path']
+
+        with Db(path) as db:
+
+            with TemporaryDirectory() as d:
+                tpm2 = Tpm2(d, path)
+
+                sobjctx, sobjauth, encobjauth, objauth = NewKeyCommandBase.new_key_init(args, db, tpm2)
+
+                tertiarypriv, tertiarypub, tertiarypubdata = self.new_key_create(args, sobjctx, sobjauth, objauth, tpm2)
+
+                NewKeyCommandBase.new_key_save(args, tertiarypriv, tertiarypub, tertiarypubdata, encobjauth, objauth, db, tpm2)
+
+@commandlet("import")
+class ImportCommand(NewKeyCommandBase):
+    '''
+    Imports a rsa key to a token within a tpm2-pkcs11 store.
     '''
 
     # adhere to an interface
     # pylint: disable=no-self-use
     def generate_options(self, group_parser):
+        group_parser.add_argument(
+            '--privkey',
+            help='Full path of the private key to be imported.\n',
+            required=True)
         group_parser.add_argument(
             '--label',
             help='The tokens label to add a key too.\n',
             required=True)
+        group_parser.add_argument(
+            '--key-label',
+            help='The label of the key imported.\n')
```

Comment:
This should default to the integer primary key from the db like addkey command.

Reply:
The super class method is doing that for both addkey and import command. if key_label is not provided it will use the id as key_label, is that what you are saying?

Reply:
group_parser.add_argument(
            '--key-label',
            help='The label of the key imported. Defaults to "id" integer value.\n')

Does this work?